albo.py

```diff
@@ -1,120 +1,61 @@
-
-# Special Headers in which is preferable to put a break line character to separate section of tg message
-break_line_headers = ["Oggetto", "Inizio pubblicazione"]
-
-ids_to_parse = cached_announcements + list(range(last_id + 1, new_id + 1))
-
-for id in ids_to_parse:
-  tr = table.find('td', text=id).parent
-  row = tr.find_all('td')
-  message = ""
-  for i, header in enumerate(headers):
-    if header in break_line_headers:
-      message += "\n"
-    message += "*" + header + "*: " + escape_char(row[i].span.string) + "\n"
-  try:
-    attachments = ["http://albo.unict.it/" + list_item['href'] for list_item in tr.find_all('a')]
-    if len(attachments) == 0:
-      if id not in cached_announcements:
-        cached_announcements.append(id)
-    else:
-      if id in cached_announcements:
-        cached_announcements.remove(id)
-      send_telegram_message(message)
-      send_telegram_attachments(attachments)
-  except ValueError as err:
-    decoded_url = json.dumps(unquote(err.args[1]))
-    document_list = re.findall(r"https?:.+?(?=\\|\")", decoded_url)
-    error_string = "```response sent: " + err.args[0] + "\n```" + "related documents:\n" + escape_char('\n'.join(document_list))
-    print (json.dumps(unquote(err.args[1]))) #Insert here bot sending error message in group
-    for dev in CHAT_ID_DEV_LIST:
-      send_telegram_message(error_string, dev)
-  #print(message)
-
-with open("last_id.txt", "w+") as f_id:
-    f_id.write(str(new_id))
-
-with open("cached_announcements.txt", "w+") as f_cached:    
-  for cached_id in cached_announcements:
-      f_cached.write(str(cached_id) +  "\n")
+
+from utils.departments import find_all_departments
+from utils.formatting import escape_char, get_formatted_message
+from utils.resources import *
+from utils.tg import TelegramBot
+
+tg_bot = TelegramBot.from_settings_file("settings.yaml")
+
+def main ():    
+    last_id = get_last_id("data/last_id.txt")
+
+    cached_announcements = get_cached_announcements("data/cached_announcements.txt")
+
+    page = requests.get("http://albo.unict.it")
+
+    if page.status_code != 200:
+        print("Impossibile accedere al sito, riprovare più tardi")
+        quit(1)
+
+    soup = BeautifulSoup(page.content, 'html.parser')
+
+    table = soup.find('div', id='boge')
+
+    new_id = int(table.find('tr').find_next_sibling().td.string)
+
+    headers = [header.string for header in table.find('tr').find_all("td")]
+
+    ids_to_parse = cached_announcements + list(range(last_id + 1, new_id + 1))
+
+    for id in ids_to_parse:
+        tr = table.find('td', text=id).parent
+        row = tr.find_all('td')
+
+        message = get_formatted_message(row, headers)
+        tags = ' '.join(["*[" + department + "]*" for department in find_all_departments(message)])
+        message = tags + "\n\n" + message
+        
+        try:
+            attachments = ["http://albo.unict.it/" + list_item['href'] for list_item in tr.find_all('a')]
```

Comment:
wrap this `http://albo.unict.it/` into a constant like `ALBO_UNICT_URL` hoping one day they'll use HTTPS

replace this magic string into the constant and make the same above where is used this string.